package-test.yaml

```diff
@@ -0,0 +1,80 @@
+tasks:
+  - task: docker.Version
+    configFilePath: scripts/configure_dev-deps.sh
+
+  - task: shell.docker.Ensure
+    name: package
+    image: algorand/mule-linux-ubuntu
+    version: '{{ docker.Version.outputs.version }}'
+    dockerFilePath: docker/build/mule.ubuntu.Dockerfile
+    dependencies: docker.Version
+
+  - task: docker.Make
+    name: package-test-deb
+    docker:
+      image: algorand/mule-linux-ubuntu
+      version: '{{ docker.Version.outputs.version }}'
+      workDir: /projects/go-algorand
+      env: [
+        AWS_ACCESS_KEY_ID=$AWS_ACCESS_KEY_ID,
+        AWS_SECRET_ACCESS_KEY=$AWS_SECRET_ACCESS_KEY
+      ]
+      volumes: [ /var/run/docker.sock:/var/run/docker.sock ]
```

Comment:
I'm confused, is this make target starting it's own docker container? If so could this be split into two separate mule tasks, one using this docker image and another using the image of the container it's starting up? This way mule handles all the docker runs, so that way mule will know to clean them up afterward. Also this may work on our jenkins agents, but I'm not 100% sure it will

Reply:
Yes, you must have missed my comments in the that describe the ticket description above.

I don't think that what you're proposing is a good strategy for this task.  The main docker container is spinning up at least four other docker images to do test the packages, and other distros may (probably?) be added.

If we could define our tasks using loops like in a vagrantfile or ansible scripts, that would make what I just said moot.  Otherwise, it's way too much boilerplate when it can be handled as I've implemented it here.

Reply:
I'm okay with this for now, but it may need to be re-tested on the Jenkins/EKS cluster, since the agents are not native hosts.